File: app/src/main/java/org/metplus/curriculum/useCases/UserTryToLogin.java
package org.metplus.curriculum.useCases;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;

public class UserTryToLogin {
    private static Logger logger = LoggerFactory.getLogger(UserTryToLogin.class);

    @Value("${backend.admin.password}")
    private String password;

    @Value("${backend.admin.username}")
    private String username;

    public UserTryToLogin(String username, String password) {
        this.username = username;
        this.password = password;
    }

    public boolean canUserLogin(String username, String password) {
        logger.trace("canUserLogin({}, {})", username, password);
        if(username.equals(this.username) && password.equals(this.password)) {
            logger.info("User '{}' login was successfull", username);
            return true;
        }
        logger.info("Unable to login user '{}'", username);
        return false;
    }
}

File: web/src/main/kotlin/org/metplus/cruncher/web/security/useCases/UserTryToLogin.kt
package org.metplus.cruncher.web.security.useCases


import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value

class UserTryToLogin(@field:Value("\${backend.admin.username}")
                     private val username: String?, @field:Value("\${backend.admin.password}")
                     private val password: String?) {

    fun canUserLogin(username: String?, password: String?): Boolean {
        logger.trace("canUserLogin({}, {})", username, password)
        if (isUsernameCorrect(username) && isPasswordCorrect(password)) {
            logger.info("User '{}' login was successfull", username)
            return true
        }
        logger.info("Unable to login user '{}'", username)
        return false
    }

    private fun isPasswordCorrect(password: String?): Boolean {
        return password != null && password == this.password
    }

    private fun isUsernameCorrect(username: String?): Boolean {
        return username != null && username == this.username
    }

    companion object {
        private val logger = LoggerFactory.getLogger(UserTryToLogin::class.java)
    }
}
